version: "3.8"

services:
  api:
    image: similarium-rs:test
    ports:
      - "8080:8080"
    volumes:
      - .:/app
    environment:
      - RUST_LOG=${RUST_LOG}
      - SLACK_BOT_TOKEN=${SLACK_BOT_TOKEN}
      - SLACK_CLIENT_ID=${SLACK_CLIENT_ID}
      - SLACK_CLIENT_SECRET=${SLACK_CLIENT_SECRET}
      - HOST=0.0.0.0
      - DATABASE_URL=postgres://${DB_USERNAME}:${DB_PASSWORD}@db:5432/${DB_NAME}
    depends_on:
      db:
        condition: service_healthy

  db:
    image: tensorchord/pgvecto-rs:latest
    environment:
      - POSTGRES_USER=${DB_USERNAME}
      - POSTGRES_PASSWORD=${DB_PASSWORD}
      - POSTGRES_DB=${DB_NAME}
    healthcheck:
      test: ["CMD", "pg_isready", "-U", "${DB_USERNAME}"]
      interval: 10s
      timeout: 5s
      retries: 5
